### ISSUE TITLE
Add output plugin to support Apache IoTDB

### ISSUE THREAD

**Issue by citrusreticulata** [influxdata/telegraf#11561]
### Proposal:

I need an output plugin to support Apache IoTDB. And I'd like to be a contributor, **all the code has been completed**. 
What I want to do now is to submit a qualified PR.

### Current behavior:

Telegraf does not support Apache IoTDB.

### Desired behavior:

Telegraf supports Apache IoTDB as an output target.

### Use case: 

Apache IoTDB (Database for Internet of Things) is an IoT native database with high performance for data management and analysis, deployable on the edge and the cloud. I'm glad to see that Telegraf will support it.

Apache IoTDB website: <https://iotdb.apache.org>
Apache IoTDB Github: <https://github.com/apache/iotdb>

### PR submitted

A PR has been submitted. #11557

**Comment by citrusreticulata** [comment]
I'm glad to see that most of tests passed. But **run-readme-linter** failed because the CI recognized language option of readme as the first section. Of course, it was short, so the test failed according to its minimum word limit. 
This is a really interesting thing. I think a strict check is very good, but I suggest checking first few sections instead.

Then I changed the first section of readme, so the test passed.

Up to now, all tests have passed. 

Could you please take the time to review the code?
Thanks!

**Comment by citrusreticulata** [comment]
A test on windows failed. But the error occurred in another plugin:
```
=== Failed
=== FAIL: plugins/inputs/statsd TestCachesExpireAfterMaxTTL (unknown)
coverage: 78.9% of statements
panic: test timed out after 10m0s

goroutine 15 [running]:
testing.(*M).startAlarm.func1()
	C:/Program Files/Go/src/testing/testing.go:2029 +0xbb
created by time.goFunc
	C:/Program Files/Go/src/time/sleep.go:176 +0x48

goroutine 1 [chan receive, 10 minutes]:
testing.(*T).Run(0xc0003f8820, {0x176023a, 0x1b}, 0x17968b8)
	C:/Program Files/Go/src/testing/testing.go:1487 +0x750
testing.runTests.func1(0x0?)
	C:/Program Files/Go/src/testing/testing.go:1839 +0x9a
testing.tRunner(0xc0003f8820, 0xc0002dfb68)
	C:/Program Files/Go/src/testing/testing.go:1439 +0x214
testing.runTests(0xc0003d0dc0?, {0x1cab7c0, 0x2c, 0x2c}, {0x40?, 0x1ccf212a8a8?, 0x1cb4a60?})
	C:/Program Files/Go/src/testing/testing.go:1837 +0x7e5
testing.(*M).Run(0xc0003d0dc0)
	C:/Program Files/Go/src/testing/testing.go:1719 +0xa72
main.main()
	_testmain.go:199 +0x3aa

goroutine 239 [sync.Cond.Wait, 10 minutes]:
sync.runtime_notifyListWait(0xc00014c290, 0x0)
	C:/Program Files/Go/src/runtime/sema.go:513 +0x146
sync.(*Cond).Wait(0xc00014c280)
	C:/Program Files/Go/src/sync/cond.go:56 +0xa5
github.com/influxdata/telegraf/testutil.(*Accumulator).Wait(0xc000538770, 0x3)
	C:/Users/circleci/project/testutil/accumulator.go:332 +0x1b7
github.com/influxdata/telegraf/plugins/inputs/statsd.TestCachesExpireAfterMaxTTL(0xc000487d40?)
	C:/Users/circleci/project/plugins/inputs/statsd/statsd_test.go:1050 +0x67a
testing.tRunner(0xc000487d40, 0x17968b8)
	C:/Program Files/Go/src/testing/testing.go:1439 +0x214
created by testing.(*T).Run
	C:/Program Files/Go/src/testing/testing.go:1486 +0x725
```

For more detail please see [here](https://app.circleci.com/pipelines/github/influxdata/telegraf/11858/workflows/a2294c69-fc8c-4417-be95-ddbb3281c5ef/jobs/193246).

Maybe this is a random error with a small probability.

**Comment by srebhan** [comment]
@citrusreticulata I think this can be closed now that #11557 is merged!?

**Comment by citrusreticulata** [comment]
All right! Thanks!
